src/plots/smith/smith.js

```diff
@@ -0,0 +1,1115 @@
+        var updateObj = {};
+        computeZoomUpdates(updateObj);
+
+        dragBox.showDoubleClickNotifier(gd);
+
+        Registry.call('_guiRelayout', gd, updateObj);
+    }
+
+    function computeZoomUpdates(update) {
+        var rl = radialAxis._rl;
+        var m = (rl[1] - rl[0]) / (1 - innerRadius / radius) / radius;
+        var newRng = [
+            rl[0] + (r0 - innerRadius) * m,
+            rl[0] + (r1 - innerRadius) * m
+        ];
+        update[_this.id + '.realaxis.range'] = newRng;
+    }
+
+    function zoomClick(numClicks, evt) {
+        var clickMode = gd._fullLayout.clickmode;
+
+        dragBox.removeZoombox(gd);
+
+        // TODO double once vs twice logic (autorange vs fixed range)
+        if(numClicks === 2) {
+            var updateObj = {};
+            for(var k in _this.viewInitial) {
+                updateObj[_this.id + '.' + k] = _this.viewInitial[k];
+            }
+
+            gd.emit('plotly_doubleclick', null);
+            Registry.call('_guiRelayout', gd, updateObj);
+        }
+
+        if(clickMode.indexOf('select') > -1 && numClicks === 1) {
+            selectOnClick(evt, gd, [_this.xaxis], [_this.yaxis], _this.id, dragOpts);
+        }
+
+        if(clickMode.indexOf('event') > -1) {
+            Fx.click(gd, evt, _this.id);
+        }
+    }
+
+    dragOpts.prepFn = function(evt, startX, startY) {
+        var dragModeNow = gd._fullLayout.dragmode;
+
+        var bbox = mainDrag.getBoundingClientRect();
+        gd._fullLayout._calcInverseTransform(gd);
+        var inverse = gd._fullLayout._invTransform;
+        scaleX = gd._fullLayout._invScaleX;
+        scaleY = gd._fullLayout._invScaleY;
+        var transformedCoords = Lib.apply3DTransform(inverse)(startX - bbox.left, startY - bbox.top);
+        x0 = transformedCoords[0];
+        y0 = transformedCoords[1];
+
+        // need to offset x/y as bbox center does not
+        // match origin for asymmetric polygons
+        if(vangles) {
+            var offset = helpers.findPolygonOffset(radius, sectorInRad[0], sectorInRad[1], vangles);
+            x0 += cxx + offset[0];
+            y0 += cyy + offset[1];
+        }
+
+        switch(dragModeNow) {
+            case 'zoom':
+                if(vangles) {
+                    dragOpts.moveFn = zoomMoveForPolygons;
+                } else {
+                    dragOpts.moveFn = zoomMove;
+                }
+                dragOpts.clickFn = zoomClick;
+                dragOpts.doneFn = zoomDone;
+                zoomPrep(evt, startX, startY);
+                break;
+            case 'select':
+            case 'lasso':
+                prepSelect(evt, startX, startY, dragOpts, dragModeNow);
+                break;
+        }
+    };
+
+    mainDrag.onmousemove = function(evt) {
+        Fx.hover(gd, evt, _this.id);
+        gd._fullLayout._lasthover = mainDrag;
+        gd._fullLayout._hoversubplot = _this.id;
+    };
+
+    mainDrag.onmouseout = function(evt) {
```

Comment:
These drag functions could be removed. No?

Reply:
Unfortunately, they can't, because they control the hover interactions. But much of the other code in this section is expendable; see 6b0f08f867fc70bd17a4c705c9c8b64ca74ecb3a.